guides/source/active_model_basics.md

```diff
@@ -38,82 +66,389 @@ class EmailContact
 
-  attribute_method_prefix 'reset_'
-  attribute_method_suffix '_highest?'
-  define_attribute_methods 'age'
+  attribute_method_affix prefix: "reset_", suffix: "_to_default!"
+  attribute_method_prefix "first_", "last_"
+  attribute_method_suffix "_short?"
+
+  define_attribute_methods "name"
 
-  attr_accessor :age
+  attr_accessor :name
 
   private
-    def reset_attribute(attribute)
-      send("#{attribute}=", 0)
+    # Attribute method call for 'first_name'
+    def first_attribute(attribute)
+      public_send(attribute).split.first
+    end
+
+    # Attribute method call for 'last_name'
+    def last_attribute(attribute)
+      public_send(attribute).split.last
+    end
+
+    # Attribute method call for 'name_short?'
+    def attribute_short?(attribute)
+      public_send(attribute).length < 5
     end
 
-    def attribute_highest?(attribute)
-      send(attribute) > 100
+    # Attribute method call 'reset_name_to_default!'
+    def reset_attribute_to_default!(attribute)
+      public_send("#{attribute}=", "Default Name")
     end
 end
 ```
 
 ```irb
 irb> person = Person.new
-irb> person.age = 110
-irb> person.age_highest?
-=> true
-irb> person.reset_age
-=> 0
-irb> person.age_highest?
+irb> person.name = "Jane Doe"
+
+irb> person.first_name
+=> "Jane"
+irb> person.last_name
+=> "Doe"
+
+irb> person.name_short?
+=> false
+
+irb> person.reset_name_to_default!
+=> "Default Name"
+```
+
+If you call a method that is not defined, it will raise a `NoMethodError` error.
+
+#### Method: `alias_attribute`
+
+`ActiveModel::AttributeMethods` provides aliasing of attribute methods using
+`alias_attribute`.
+
+The example below creates an alias attribute for `name` called `full_name`. They
+return the same value, but the alias `full_name` better reflects that the
+attribute includes a first name and last name.
+
+```ruby
+class Person
+  include ActiveModel::AttributeMethods
+
+  attribute_method_suffix "_short?"
+  define_attribute_methods :name
+
+  attr_accessor :name
+
+  alias_attribute :full_name, :name
+
+  private
+    def attribute_short?(attribute)
+      public_send(attribute).length < 5
+    end
+end
+```
+
+```irb
+irb> person = Person.new
+irb> person.name = "Joe Doe"
+irb> person.name
+=> "Joe Doe"
+
+# `full_name` is the alias for `name`, and returns the same value
+irb> person.full_name
+=> "Joe Doe"
+irb> person.name_short?
+=> false
+
+# `full_name_short?` is the alias for `name_short?`, and returns the same value
+irb> person.full_name_short?
 => false
 ```
 
 ### Callbacks
 
-`ActiveModel::Callbacks` gives Active Record style callbacks. This provides an
-ability to define callbacks which run at appropriate times.
-After defining callbacks, you can wrap them with before, after, and around
-custom methods.
+[`ActiveModel::Callbacks`](https://api.rubyonrails.org/classes/ActiveModel/Callbacks.html)
+gives plain Ruby objects [Active Record style
+callbacks](https://guides.rubyonrails.org/active_record_callbacks.html). The
```

Comment:
Minor: Let's just link to the guide without the full URL: (so local generated guides point to other local ones, etc.)

```suggestion
callbacks](active_record_callbacks.html). The
```